File: projects/localserver/commons/src/main/java/com/robinfood/localserver/commons/entities/orders/OrderDetailProductGroupPortionEntity.java
package com.robinfood.localserver.commons.entities.orders;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@AllArgsConstructor
@Builder
@Data
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class OrderDetailProductGroupPortionEntity {

    private Boolean addition;

    private OrderDetailChangedPortionEntity changedPortion;

    private BigDecimal discount;

    private Integer quantityFree;

    private Long id;

    private String name;

    private Double price;

    private Long parentId;

    private Integer quantity;

    private String sku;

    private Long units;

    private Double weight;
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/entities/OrderDetailProductGroupPortionEntity.kt
package com.robinfood.core.entities

import java.math.BigDecimal

data class OrderDetailProductGroupPortionEntity(
        val addition: Boolean?,
        val changedPortion: OrderDetailChangedPortionEntity?,
        val discount: BigDecimal?,
        val free: Int?,
        val id: Long?,
        val name: String?,
        val parentId: Long?,
        val price: BigDecimal?,
        val quantity: Int?,
        val sku: String?,
        val units: Long?,
        val weight: Double?
)
